sdcm/nemesis.py

```diff
@@ -496,28 +496,51 @@ def disrupt_nodetool_refresh(self, big_sstable=True, skip_download=False):
             sstable_url = 'https://s3.amazonaws.com/scylla-qa-team/refresh_nemesis/keyspace1.standard1.100M.tar.gz'
             sstable_file = '/tmp/keyspace1.standard1.100M.tar.gz'
             sstable_md5 = '9c5dd19cfc78052323995198b0817270'
+            keys_num = 1000
         else:
             sstable_url = 'https://s3.amazonaws.com/scylla-qa-team/refresh_nemesis/keyspace1.standard1.tar.gz'
             sstable_file = "/tmp/keyspace1.standard1.tar.gz"
             sstable_md5 = 'c033a3649a1aec3ba9b81c446c6eecfd'
-        if not skip_download:
-            key_store = KeyStore()
-            creds = key_store.get_scylladb_upload_credentials()
-            remote_get_file(self.target_node.remoter, sstable_url, sstable_file,
-                            hash_expected=sstable_md5, retries=2,
-                            user_agent=creds['user_agent'])
+            keys_num = 501000
 
         self.log.debug('Prepare keyspace1.standard1 if it does not exist')
-        self._prepare_test_table(ks='keyspace1')
+        self._prepare_test_table(ks='keyspace1', table='standard1')
         result = self.target_node.run_nodetool(sub_cmd="cfstats", args="keyspace1.standard1")
-        if result is not None and result.exit_status == 0:
-            result = self.target_node.remoter.run("sudo ls -t /var/lib/scylla/data/keyspace1/")
+
+        def do_refresh(node):
+            if not skip_download:
+                key_store = KeyStore()
+                creds = key_store.get_scylladb_upload_credentials()
+                # Download the sstable files from S3
+                remote_get_file(node.remoter, sstable_url, sstable_file,
+                                hash_expected=sstable_md5, retries=2,
+                                user_agent=creds['user_agent'])
+            result = node.remoter.run("sudo ls -t /var/lib/scylla/data/keyspace1/")
             upload_dir = result.stdout.split()[0]
-            self.target_node.remoter.run('sudo tar xvfz {} -C /var/lib/scylla/data/keyspace1/{}/upload/'.format(
+            node.remoter.run('sudo tar xvfz {} -C /var/lib/scylla/data/keyspace1/{}/upload/'.format(
                 sstable_file, upload_dir))
-            self.target_node.run_nodetool(sub_cmd="refresh", args="-- keyspace1 standard1")
-            cmd = "select * from keyspace1.standard1 where key=0x32373131364f334f3830"
-            result = self.target_node.run_cqlsh(cmd)
+            self.log.debug(f'Loading {keys_num} keys to {node.name} by refresh')
+            node.run_nodetool(sub_cmd="refresh", args="-- keyspace1 standard1")
+
+        if result is not None and result.exit_status == 0:
+            query_cmd = "SELECT * FROM keyspace1.standard1 LIMIT 1"
+            result = self.target_node.run_cqlsh(query_cmd)
+            # Add columns if they don't exist, otherwise the sstable files won't loaded by cluster by refresh
+            for col in ['C1', 'C2', 'C3', 'C4']:
```

Comment:
How do we test refresh in 200gb longevity which only has 1 column in keyspace1.standard1?
The backup sstable files have default 5 columns.

If we load sstable files that have mismatched schema, it might break the test.

Reply:
For now, till issue https://github.com/scylladb/scylla/issues/6617 will be fixed, we should have another set (or sets) of files with other schema's (like this one column).
When 6617 will be fixed, we will remove this set of files and have this as a negative test to make sure that when refresh gets files that doesn't match the schema it fails.

For now, please create another set of files for refresh nemesis and check which schema to load inside the nemesis (please note that we have other longevities with more than 5 columns but it seems that in that case there is no need for special handling).

Reply:
ok

Reply:
did you do it?

Reply:
Yes